bm-persona/Libraries/LibraryDetailViewController.swift

```diff
@@ -49,156 +40,101 @@ class LibraryDetailViewController: SearchDrawerViewController {
+            self.updateScrollView()
+        })
+        let openTimesCard = self.openTimesCard!
+        scrollContent.addSubview(openTimesCard)
+        openTimesCard.topAnchor.constraint(equalTo: overviewCard.bottomAnchor, constant: kViewMargin).isActive = true
+        openTimesCard.leftAnchor.constraint(equalTo: scrollContent.leftAnchor).isActive = true
+        openTimesCard.rightAnchor.constraint(equalTo: scrollContent.rightAnchor).isActive = true
     }
     
-//    func setUpTrafficCard() {
-//        let overviewCard = CardView()
-//        overviewCard.layoutMargins = kCardPadding
-//        scrollView.addSubview(overviewCard)
-//        overviewCard.translatesAutoresizingMaskIntoConstraints = false
-//        overviewCard.topAnchor.constraint(equalTo: scrollView.layoutMarginsGuide.topAnchor, constant: 400).isActive = true
-//        overviewCard.leftAnchor.constraint(equalTo: scrollView.layoutMarginsGuide.leftAnchor).isActive = true
-//        overviewCard.rightAnchor.constraint(equalTo: scrollView.layoutMarginsGuide.rightAnchor).isActive = true
-//        overviewCard.heightAnchor.constraint(equalToConstant: 200).isActive = true
-//    }
-//
-    
     func setUpBookButton() {
-        bookButton.isUserInteractionEnabled = true
-        bookButton.layoutMargins = kCardPadding
-        scrollView.addSubview(bookButton)
-        bookButton.backgroundColor =  Color.eventAcademic
-        bookButton.layer.cornerRadius = 10
-        bookButton.layer.shadowRadius = 5
-        bookButton.layer.shadowOpacity = 0.25
-        bookButton.layer.shadowOffset = .zero
-        bookButton.layer.shadowColor = UIColor.black.cgColor
-        bookButton.layer.shadowPath = UIBezierPath(rect: bookButton.layer.bounds.insetBy(dx: 4, dy: 4)).cgPath
-        bookButton.translatesAutoresizingMaskIntoConstraints = false
-        bookButton.topAnchor.constraint(equalTo: hoursCard.layoutMarginsGuide.bottomAnchor, constant: kViewMargin).isActive = true
-        bookButton.leftAnchor.constraint(equalTo: scrollView.layoutMarginsGuide.leftAnchor).isActive = true
-        bookButton.rightAnchor.constraint(equalTo: scrollView.layoutMarginsGuide.rightAnchor).isActive = true
+        scrollContent.addSubview(bookButton)
+        bookButton.topAnchor.constraint(equalTo: openTimesCard?.bottomAnchor ?? overviewCard.bottomAnchor, constant: kViewMargin).isActive = true
+        bookButton.leftAnchor.constraint(equalTo: scrollContent.leftAnchor).isActive = true
+        bookButton.rightAnchor.constraint(equalTo: scrollContent.rightAnchor).isActive = true
+        bookButton.bottomAnchor.constraint(equalTo: scrollContent.bottomAnchor).isActive = true
         bookButton.heightAnchor.constraint(equalToConstant: 50).isActive = true
-        bookButton.setTitle("Book a Study Room >", for: .normal)
-        bookButton.titleLabel!.font = Font.semibold(14)
-        bookButton.titleLabel!.textColor = .white
-        
     }
     
     func setUpScrollView() {
```

Comment:
I'm a bit confused about this hierarchy. So if I'm understanding this correctly, it goes `scrollContainer > scrollView > scrollContent`, with an additional `8px` padding between the `scrollView` and its content.

Could you explain a bit about the purpose of `scrollContainer` and why you chose a non-zero `scrollView.contentInset`?

Reply:
so I wanted a margin of 16 between the scroll view and the card so that the user can move the drawer by dragging the left or right margin (otherwise the scroll would register as scrolling the scroll view). but I was running into some problem (I think the scroll view wasn't scrolling or something? don't remember exactly) where the solution was that the scroll view needed to be constrained with 0 margins to its superview. so I used scrollContainer to get the margins and constrained the scroll view to it with 0 margins. the contentInset is nonzero because otherwise the shadows would get cut off.

might be a better way to do this but this is the best I could come up with after running into some issues since I haven't used scroll views much.